Source/Moya.swift

```diff
@@ -65,12 +87,16 @@ public class MoyaProvider<Target: TargetType> {
     }
     
     /// Returns an Endpoint based on the token, method, and parameters by invoking the endpointsClosure.
-    public func endpoint(token: Target) -> Endpoint<Target> {
+    public func endpoint(token: TargetType) -> Endpoint<TargetType> {
         return endpointClosure(token)
     }
     
     /// Designated request-making method. Returns a Cancellable token to cancel the request later.
     public func request(target: Target, completion: Moya.Completion) -> Cancellable {
+        guard let target = target.resource else {
```

Comment:
Could you expand on the use of `.resource` here, I'm not sure I understand it or `ServiceType` but I'd like to know more!